src/ledger/LedgerTestUtils.cpp

```diff
@@ -102,26 +104,34 @@ makeValid(TrustLineEntry& tl)
     clampLow<int64_t>(0, tl.balance);
     clampLow<int64_t>(1, tl.limit);
     clampHigh<int64_t>(tl.limit, tl.balance);
+    tl.flags = tl.flags & MASK_TRUSTLINE_FLAGS;
 }
 void
 makeValid(OfferEntry& o)
 {
+    if (o.offerID > INT64_MAX)
+    {
+        o.offerID -= INT64_MAX;
+    }
     o.selling.type(ASSET_TYPE_CREDIT_ALPHANUM4);
     strToAssetCode(o.selling.alphaNum4().assetCode, "CAD");
 
     o.buying.type(ASSET_TYPE_CREDIT_ALPHANUM4);
     strToAssetCode(o.buying.alphaNum4().assetCode, "EUR");
 
-    clampLow<int64_t>(0, o.amount);
-    clampLow(0, o.price.n);
+    if (o.amount < 0)
+    {
+        o.amount = -o.amount;
+    }
+    clampLow(1, o.price.n);
```

Comment:
Using clampLow instead of absolute value reduces the entropy of the overall price, giving a significant probability that price = price.n / price.d == 1.

Reply:
fixed